Release notes for 7.12.0 (#12683)
elastic/logstash: docs/static/releasenotes.asciidoc

compatibility mode] is enabled these fields are moved to ECS-compatible
namespace.

===== JDK 15 support

Logstash introduces support for JDK 15!  You need to update settings in
`jvm.options` and `log4j2.properties` if:

* you are upgrading from Logstash 7.11.x (or earlier) to 7.12 or later, AND
* you are using JDK 15 or later.

Unless both of these conditions apply, you don't need to adjust settings because
of the upgrade.  See <<jdk15-upgrade,Using JDK 15>> for more information.

===== Conditional settings for JVM versions

We've added support for conditional settings and behavior, dependent on the JVM
version. Now you can configure different settings for different JVM versions.
Here is an example from the default `jvm.options` file.

Example:
```
## GC configuration
8-13:-XX:+UseConcMarkSweepGC
8-13:-XX:CMSInitiatingOccupancyFraction=75
8-13:-XX:+UseCMSInitiatingOccupancyOnly
```
This example sets garbage collection (GC) values for JDK 8-13 only. Those
settings don't apply to JVM 14 and above.

This feature is available for any setting in the `jvm.options` file, and aligns
more closely with the {es} implementation of jvm settings.

===== ARM64 support for Linux (beta)

> Support for 64-bit ARM architectures on Linux is now in beta, with downloadable artifacts and docker images available

Reviewer: ```suggestion
Support for 64-bit ARM architectures on Linux is now in beta, with downloadable artifacts and docker images available.
```

==== Performance improvements and notable issues fixed

**Pipeline Loading and Monitoring improvements**

We've made changes to start the webserver that exposes the Logstash metrics API earlier in the startup process.
For slow starting pipelines, this would cause error messages to appear in the Logstash logs, and cause delays to
the availability of the metrics API. https://github.com/elastic/logstash/pull/12571[#12571]

**Windows Startup Fixes**

We've fixed an issue where Logstash would crash when attempting to start using the bundled JDK when Logstash was located
in a folder where the folder name contained spaces https://github.com/elastic/logstash/pull/12585[#12585]

**Certificate Verification with Internal Monitoring**

> A bug in the monitoring pipeline was causing it to always start its Elasticsearch Output Plugin with certificate
> verification disabled, which meant that Logstash Internal Monitoring has been sending _monitoring_ metadata (such as
> pipeline throughput metrics) to Elasticsearch without verifying the recipient. This bug did _NOT_ affect instances of
> the Elasticsearch Output Plugin in other pipelines. https://github.com/elastic/logstash/pull/12749[#12749]

Reviewer: ```suggestion
We fixed a bug in the monitoring pipeline that caused it to pass monitoring data
to {es} with certificate verification disabled. {ls} internal monitoring had
been sending monitoring metadata (such as pipeline throughput metrics) to {es}
without verifying the recipient.
https://github.com/elastic/logstash/pull/12749[#12749]
```

*Http Input - 3.3.7*

* Feat: improved error handling/logging/unwraping https://github.com/logstash-plugins/logstash-input-http/pull/133[#133]

*Redis Input - 3.6.0*

* Remove ruby pipeline dependency. Starting from Logstash 8, Ruby execution engine
is not available. All pipelines should use Java pipeline
https://github.com/logstash-plugins/logstash-input-redis/pull/84[#84]

*Syslog Input - 3.4.5*

* Added support for listening on IPv6 addresses

*Tcp Input - 6.0.7*

* Fix: reduce error logging (to info level) on connection resets https://github.com/logstash-plugins/logstash-input-tcp/pull/168[#168]
* Refactor: only patch Socket classes once (on first input)
* Refactor: use a proper log4j logger (in Java to avoid surprises when unwrapping `LogStash::Logging::Logger`)

*Udp Input - 3.4.0*

* Added ECS compatibility mode (`disabled` and `v1`) to rename ip source address in a ECS compliant name https://github.com/logstash-plugins/logstash-input-udp/pull/50[#50]
* Fixed integration tests for IPv6 downgrading Docker to version 2.4 https://github.com/logstash-plugins/logstash-input-udp/pull/51[#51]

*Kafka Integration - 10.7.1*

> * Fix: dropped usage of SHUTDOWN event deprecated since Logstash 5.0 https://github.com/logstash-plugins/logstash-integration-kafka/issue/71[#71]

Reviewer: ```suggestion
* Fix: dropped usage of SHUTDOWN event deprecated since Logstash 5.0 https://github.com/logstash-plugins/logstash-integration-kafka/pull/71[#71]
```

Reviewer: Fix broken link
